Mlem/App/Views/Shared/Images/Core/DynamicImageView.swift

```diff
@@ -56,5 +57,58 @@ struct DynamicImageView: View {
             .onChange(of: loader.loading, initial: true) { loadingPref = loader.loading }
             .preference(key: ImageLoadingPreferenceKey.self, value: loadingPref)
             .task(loader.load)
+            .contextMenu {
+                if let uiImage = loader.uiImage {
+                    Button {
+                        Task {
+                            await saveImage()
+                        }
+                    } label: {
+                        Label(String(localized: "Save Image"), systemImage: Icons.import)
+                    }
+                    
+                    ShareLink(item: Image(uiImage: uiImage), preview: .init("photo", image: Image(uiImage: uiImage)))
+                    
+                    if let url = loader.url {
+                        Button {
+                            Task {
+                                await showQuickLook(url: url)
+                            }
+                        } label: {
+                            Label(String(localized: "Quick Look"), systemImage: Icons.imageDetails)
+                        }
+                    }
+                }
+            }
+            .quickLookPreview($quickLookUrl)
+    }
+    
+    func saveImage() async {
+        do {
+            let (data, _) = try await ImagePipeline.shared.data(for: .init(url: loader.url))
```

Comment:
`loader.url` can include the `thumbnail=1024` query item, which needs to be excluded somehow. iirc query items can't be removed from a `URL` directly - one option would be to convert to `URLComponents`, remove the query item then convert back again

Reply:
Good catch--I've implemented a fix that saves and opens quickLook at higher resolutions. Share is a tricky case, because it needs the image to share to be readily available--that should be easy to handle with the improved image fetching in #1244.

Reply:
I added a way to do this programatically a while ago: `navigation.shareUrl = url`. It would need adapting to accept images too, though - maybe with an enum?